### ISSUE TITLE
If queues do not exist, SC must be reinstalled to recreated the queues

### ISSUE THREAD

**Issue by ramonsmits** [Particular/ServiceControl#1051]
SC does not start after the queue was not present anymore. In my case because I reinstalled MSMQ. I had to reinstall SC for it to recreate all relevant queues.

It would be nice if the installers could be rerun.

Note: Similar to https://github.com/Particular/ServiceControl.Monitoring/issues/92

**Comment by WilliamBZA** [comment]
The installers can be rerun from the command line. I don't know if it's documented anywhere...

**Comment by fourpastmidnight** [comment]
No, it's not documented anywhere. I found this, too, by perusing the source code.

**Comment by fourpastmidnight** [comment]
Use the `-s|--setup` switch  when running `ServiceControl.exe`:

```text
ServiceControl.exe --serviceName="My Instance Name" --setup
```

HTH.
